src/hotspot/share/opto/partialEscape.hpp

```diff
@@ -317,22 +353,26 @@ class PEAState {
   void print_on(outputStream* os) const;
 #endif
 
-#ifdef ASSERT
-  void validate() const;
-#endif
+protected:
+  using ObjColorTable = ResourceHashtable<ObjID, bool, 23, /*table_size*/
+                                 AnyObj::RESOURCE_AREA, mtCompiler>;
+
+  void dfs(GraphKit* kit, ObjColorTable& color, ObjID key, VirtualState* vs, RegionNode* region, RegionNode* shim, const GrowableArray<uint>& paths);
```

Comment:
dfs? This does not tell me what the function does. Maybe `materialize_object_graph`?